practices/github-conventions.md

```diff
@@ -0,0 +1,73 @@
+# GitHub repository conventions
+
+This page describes team 
+
+## Issue labels
+
+There are a few issue labels that we use to provide key metadata in our issues.
+These are added to all `2i2c-org/` repositories, and share the same meaning across each.
+
+:::{note}
+- Only **Issue Type** is required for all issues.
+- Repositories may define their own labels in addition to the ones described here for a given category, unless otherwise noted.
+:::
+
+### Issue Type
+
+**REQUIRED**
+
+Issue type determines the kind of issue and implies what sorts of actions must be taken to close it.
+Issue types are mutually-exclusive - there may only be one issue type per issue.
+
+There are a few issue types that are defined for all repositories:
+
+- {badge}`type: enhancement, badge-info`: an improvement to be made to the repository.
+- {badge}`type: bug, badge-info`: something that is not working or incorrect in the repository.
+- {badge}`type: task, badge-info`: an action to take that is not well-categorized by enhancement/bug.
+- {badge}`type: goal, badge-info`: a high-level goal or outcome that we should work towards, with a specific completion criteria.
```

Comment:
What is the interrelationship between the goal label and the concept of deliverables we have in the board structure?

Reply:
In my mind, a goal is more abstract and will likely happen over longer periods of time. E.g., "improve our team communication practices"

However, I think we could certainly recreate the same information with a combination of "enhancement" plus tags. Maybe we should just remove it for now?

Reply:
I think it will cause confusion, so I would be +1 on removing it for now.

Reply:
Should we also remove the goal issue template from the central `.github` repo https://github.com/2i2c-org/.github/tree/main/ISSUE_TEMPLATE? Though, looking at the repo now, it looks like the `new-goal` issue template doesn't have any effect on the org repos. I believe we should have the `ISSUE_TEMPLATE` dir inside a `.github` one for it to work.

Also, since we want the issue type to be a required label, maybe we could have some simple issue templates for this kinds of issue types in the `.github` that automatically add the label.

Reply:
@GeorgianaElena good point, and agreed. Though I think that if we define custom issue templates in any other repository, they *overwrite* the `.github` ones (AKA, they aren't *added* to the `.github/` ones, they replace them all)